spec/models/active_fedora_datastream_spec.rb

```diff
@@ -3,6 +3,36 @@
 module ActiveFedora
   RSpec.describe Datastream do
 
+    describe "#tempfile" do
+      subject { described_class.new(nil, nil, controlGroup: "M") }
+      describe "when the datastream has no content" do
+        it "should raise an exception" do
+          expect { subject.tempfile { |f| puts f.path } }.to raise_error(Ddr::Models::Error)
+        end
+      end
+      describe "when the datastream has content" do
+        let(:file) { fixture_file_upload("sample.pdf", "application/pdf") }
+        before do
+          subject.content = file.read
+          subject.mimeType = file.content_type
+          allow(subject).to receive(:pid) { "test:1" }
+        end
+        describe "the yielded file" do
+          it "should have extension for the datastream media type" do
+            subject.tempfile do |f|
+              expect(f.path.end_with?(".pdf")).to be true
+            end
+          end
+          it "should have the same size as the datastream" do
+            subject.tempfile do |f|
+              puts subject.content.length
+              expect(f.size).to eq(file.size)
```

Comment:
Not sure why you puts subject.content.length in line 28.  Is that needed for the test?  Also, in line 29, would it be better to compare f.size to subject.content.length (instead of file.size)?